Confusing error when a space is inserted between comment lines

I'm not sure whether this is valid according to BED spec, but if you accidentally introduce a blank line between comments, pybedlite will refuse to parse it (I believe, but still need to check, that pybedtools _will_ accept it).

```
example_file = "# some line\n# another line\n#\t\t third line\n\n#\n# Another comment\n# CHROM START END NAME\n# chr5\t1\t500"
with open("example.bed", "w") as fp:
    fp.write(example_file)
list(pybedlite.BedSource("example.txt"))
```

Produces this traceback:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../site-packages/pybedlite/bed_source.py", line 128, in __iter__
    + f"fields. On line {i} in {self._path} had only {len(fields)} fields"
AssertionError: BED records must conform to specifications, which requires at least 3 input fields. On line 3 in example.txt had only 1 fields
```

I'm totally open to idea that this is expected behavior, but it's actually a little nuanced to figure out what's going on.

Also, it appears that pybedtools is printing the line off by 1? (since it's actually line 4 in the file that has 1 field, line 3 is the line BEFORE the space).

--- a/README.md
+++ b/README.md
@@ -60,4 +60,7 @@ export CFLAGS="-stdlib=libc++"
 [conda-link]:  https://docs.conda.io/en/latest/miniconda.html
 
 ## Checking the Build
+### Run all checks with:
+```bash
 ./ci/check.sh
+```

--- a/README.md
+++ b/README.md
@@ -58,3 +58,6 @@ export CFLAGS="-stdlib=libc++"
 
 [poetry-link]: https://github.com/python-poetry/poetry
 [conda-link]:  https://docs.conda.io/en/latest/miniconda.html
+
+## Checking the Build
+./ci/check.sh

--- a/pybedlite/bed_source.py
+++ b/pybedlite/bed_source.py
@@ -122,6 +122,8 @@ class BedSource(ContextManager, Iterable[BedRecord]):
             # Skip header lines
             if line.startswith("#") or line.startswith("browser") or line.startswith("track"):
                 continue
+            if line.strip() == "":
+                continue
             fields = line.strip().split("\t")
             assert len(fields) >= 3, (
                 "BED records must conform to specifications, which requires at least 3 input "

--- a/pybedlite/tests/test_pybedlite.py
+++ b/pybedlite/tests/test_pybedlite.py
@@ -61,6 +61,9 @@ def bed_records() -> List[BedRecord]:
 
 
 SNIPPET_BED = """\
+# Test header, with a line with whitespace below it.
+
+# More header
 1	100	150	test_record1	100	+	100	100	0,0,0	1	50	0
 1	200	300	test_record2	100	-	210	290	0,0,0	1	100	0
 2	200	300	test_record3	.	.	.	.	.	.	.	.

